unity-renderer/Assets/UIComponents/Scripts/Components/SectionSelector/SectionSelectorComponentView.cs

```diff
@@ -0,0 +1,142 @@
+    /// </summary>
+    /// <param name="index">Index of the list of sections.</param>
+    /// <returns>A specific section toggle.</returns>
+    ISectionToggle GetSection(int index);
+
+    /// <summary>
+    /// Get all the sections of the section selector.
+    /// </summary>
+    /// <returns>The list of sections.</returns>
+    List<ISectionToggle> GetAllSections();
+}
+
+public class SectionSelectorComponentView : BaseComponentView, ISectionSelectorComponentView
+{
+    [Header("Prefab References")]
+    [SerializeField] internal SectionToggle sectionToggleTemplate;
+
+    [Header("Configuration")]
+    [SerializeField] internal SectionSelectorComponentModel model;
+
+    private List<ISectionToggle> instantiatedSections = new List<ISectionToggle>();
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        Configure(model);
+    }
+
+    public void Configure(SectionSelectorComponentModel model)
+    {
+        this.model = model;
+        RefreshControl();
+    }
+
+    public override void RefreshControl()
+    {
+        if (model == null)
+            return;
+
+        SetSections(model.sections);
+    }
+
+    public void SetSections(List<SectionToggleModel> sections)
+    {
+        model.sections = sections;
+
+        RemoveAllIntantiatedSections();
+
+        for (int i = 0; i < sections.Count; i++)
+        {
+            CreateSection(sections[i], $"Section_{i}");
+        }
+    }
+
+    public ISectionToggle GetSection(int index)
+    {
+        if (index >= instantiatedSections.Count)
+            return null;
+
+        return instantiatedSections[index];
+    }
+
+    public List<ISectionToggle> GetAllSections() { return instantiatedSections; }
+
+    internal void CreateSection(SectionToggleModel newSectionModel, string name)
+    {
+        if (Application.isPlaying)
+        {
+            IntantiateSectionToggle(newSectionModel, name);
+        }
+        else
+        {
+            if (isActiveAndEnabled)
+                StartCoroutine(IntantiateSectionToggleOnEditor(newSectionModel, name));
+        }
+    }
+
+    internal void IntantiateSectionToggle(SectionToggleModel newSectionModel, string name)
+    {
+        if (sectionToggleTemplate == null)
+            return;
+
+        SectionToggle newGO = Instantiate(sectionToggleTemplate, transform);
+        newGO.name = name;
+        newGO.SetInfo(newSectionModel);
+        newGO.gameObject.SetActive(true);
+        instantiatedSections.Add(newGO);
+    }
+
+    internal IEnumerator IntantiateSectionToggleOnEditor(SectionToggleModel sectionModel, string name)
+    {
+        yield return null;
+        IntantiateSectionToggle(sectionModel, name);
+    }
+
+    internal void RemoveAllIntantiatedSections()
+    {
+        foreach (Transform child in transform)
+        {
+            if (child.gameObject.GetInstanceID() == sectionToggleTemplate.gameObject.GetInstanceID())
+                continue;
```

Comment:
Why we need to compare using `GetInstanceID()`? By just comparing the objects should be enough.

Reply:
You're completely right, I have changed it. Thanks for the feedback! Done in: https://github.com/decentraland/unity-renderer/pull/1175/commits/6be3cfbf83aef15d4e8035d60a5346255aff76b2